src/main.rs

```diff
@@ -375,7 +390,8 @@ fn usage(is_error: bool) -> ! {
     let mut stderr = std::io::stderr();
     let out: &mut dyn std::io::Write = if is_error { &mut stderr } else { &mut stdout };
 
-    show_version(out);
+    let system = realsystem::RealSystem::new().freeze();
```

Comment:
This is pretty ugly, can't this be a parameter?

Reply:
The correct answer is that show_version should always print the true version, never the version from the systemapi object.  So just remove this parameterization.